client/ayon_usd/plugins/publish/integrate_usd_output_processor_remap_relative.py

```diff
@@ -0,0 +1,133 @@
+        # For those we'll log a warning.
+        if not HAS_USD_LIBS:
+            self.log.warning(
+                "Unable to process USD files to relative paths because "
+                "`pxr` USD libraries could not be imported.")
+            return
+
+        # For each USD representation, process the file.
+        for representation in instance.data.get("representations", []):
+            representation: dict
+
+            if representation.get("name") != "usd":
+                continue
+
+            # Get expected publish path
+            published_path = get_instance_expected_output_path(
+                instance,
+                representation_name=representation["name"],
+                ext=representation.get("ext")
+            )
+            published_path_root = os.path.dirname(published_path)
+
+            # Process all files of the representation
+            staging_dir: str = representation.get(
+                "stagingDir", instance.data.get("stagingDir"))
+
+            # Process single file or sequence of the representation
+            if isinstance(representation["files"], str):
+                # Single file
+                fname: str = representation["files"]
+                path = os.path.join(staging_dir, fname)
+                self.process_usd(path, start=published_path_root)
+            else:
+                # Sequence
+                for fname in representation["files"]:
+                    path = os.path.join(staging_dir, fname)
+                    self.process_usd(path, start=published_path_root)
+
+        # Some instance may have additional transferred files which
+        # themselves are not a representation. For those we need to look in
+        # the `instance.data["transfers"]`
+        for src, dest in instance.data.get("transfers", []):
+            if not dest.endswith(".usd"):
+                continue
+
+            # Process USD file at `src` and turn all paths relative to
+            # the `dest` path the file will end up at.
+            dest_root = os.path.dirname(dest)
+            self.process_usd(src, start=dest_root)
+
+    def process_usd(self, usd_path, start):
+        """Process a USD layer making all paths relative to `start`"""
+        self.log.debug(f"Processing '{usd_path}'")
+        layer = Sdf.Layer.FindOrOpen(usd_path)
+
+        def modify_fn(asset_path: str):
+            """Make all absolute non-anchored paths relative to `start`"""
+            self.log.debug(f"Processing asset path: {asset_path}")
+
+            # Do not touch paths already anchored paths
+            if not os.path.isabs(asset_path):
+                return asset_path
+
+            # Do not touch paths already anchored paths
+            if asset_path.startswith(RELATIVE_ANCHOR_PREFIXES):
+                # Already anchored
+                return asset_path
+
+            # Do not touch what we know are AYON URIs
+            if asset_path.startswith(("ayon://", "ayon+entity://")):
```

Comment:
we should respect all types of urls not just the ones we defined. 
this approach is a bit wasteful interms of performance but would work. 

```py
from urllib.parse import urlparse

def is_uri(path):
    parsed = urlparse(path)
    return bool(parsed.scheme and parsed.netloc)
```

```suggestion
            if is_uri(asset_path):
```

Reply:
I've thought about that, but it wouldn't necessarily capture all. One could technically define their resolver paths in a very different manner that is not `urlparse` compliant. It could just be `ASSET|v001` or whatever for their resolver to pick it up. It's not an absolute requirement that it is urlparsable URI?

Since we haven't hit such a case that we should skip others - I've refrained from trying to make assumptions about those.

Reply:
that makes sens but then it should be possible to configure it in settings. because currently every one need to change the code and maintain there own addon to achieve what your proposing.